internal/repos/gitolite_test.go

```diff
@@ -0,0 +1,39 @@
+package repos
+
+import (
+	"testing"
+
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/httpcli"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+func TestGitoliteSource(t *testing.T) {
+	testCases := []struct {
+		name string
+	}{
+		{
+			name: "basic",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			cf, save := newClientFactoryWithOpt(t, tc.name, httpcli.ExternalTransportOpt)
+			defer save(t)
+
+			svc := &types.ExternalService{
+				Kind:   extsvc.KindGitolite,
+				Config: marshalJSON(t, &schema.GitoliteConnection{}),
+			}
+
+			_, err := NewGitoliteSource(svc, cf)
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
```

Comment:
```suggestion
func TestGitoliteSource(t *testing.T) {
	cf, save := newClientFactoryWithOpt(t, "basic", httpcli.ExternalTransportOpt)
	defer save(t)

	svc := &types.ExternalService{
		Kind:   extsvc.KindGitolite,
		Config: marshalJSON(t, &schema.GitoliteConnection{}),
	}

	_, err := NewGitoliteSource(svc, cf)
	if err != nil {
		t.Fatal(err)
	}
}
```

Not blocking, but: let's cross that bridge once we get there :)

Reply:
(Also: do we need the call to `save(t)`? We don't do any requests)

Reply:
+1 on simplifying this for now, and I think it's fine to leave the `save(t)` here since if this test grows we don't want that to be forgotten (is that a good enough reason? 😛 )

cc @mollylogue who authored the test